### PR TITLE
Support multi-buffer sends in WebAssembly WebSockets

### PR DESCRIPTION
**Pull request by BrennanConroy**
Proposed fix for https://github.com/dotnet/runtime/issues/44551

### CODE REVIEW

**Comment by davidfowl** [comment]
Memory stream.. someday we'll care about perf in the browser

**Review by lewing — APPROVED** [review]
@kjpou1 please follow up with a test for this.

**Comment by lewing** [comment]
@kg this should be on the list of the perf tests as well

File: src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs [hunk 1]
```diff
@@ -357,15 +357,22 @@ public override Task SendAsync(ArraySegment<byte> buffer, WebSocketMessageType m
...
 
-            MemoryStream writtenBuffer = _writeBuffer;
+            MemoryStream? writtenBuffer = _writeBuffer;
             _writeBuffer = null;
 
+            if (writtenBuffer is not null)
+            {
+                writtenBuffer.Write(buffer.Array!, buffer.Offset, buffer.Count);
+            }
+            buffer = writtenBuffer?.ToArray() ?? buffer;
```
Comment (lewing) [inline src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs:374]: Should be possible to store the writtenBuffer.Count and use TryGetBuffer here then fall back to ToArray ()

Comment (BrennanConroy) [reply src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs]: Good call

**Comment by halter73** [comment]
At least this no longer allocates the MemoryStream for a single-frame message, but pooling would be nice.

**Comment by lewing** [comment]
@halter73 I had already opened https://github.com/dotnet/runtime/issues/44614 to track other improvements.

**Comment by lewing** [comment]
Failures are unrelated and the FileProviders issue is resolved now.

**Pull request merged**